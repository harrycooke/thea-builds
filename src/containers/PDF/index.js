import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  CircularProgress
} from "@material-ui/core";
import { readMessages } from '../../actions/chatActions';
import { setCurrentCase } from '../../actions/caseAction';
import MyDocument from '../../components/ReactPdf/document';

class PDF extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      channel_id: '',
      currentCase: '',
      readedMessages: [],
      isLoading: true,
      readedMessagesWasSetted: false,
      currentCaseWasSetted: false
    }
    this.callDownloadPDF = this.callDownloadPDF.bind(this)
  }

  UNSAFE_componentWillMount() {
    const { channel_id } = this.props;
    this.setState({ channel_id, isLoading: true });
  }

  componentDidMount() {
    if(this.props && typeof this.props.onRef === 'function'){
      this.props.onRef(this);
    }
    this.setState({
      userdata: this.props.userdata
    });
    const { channel_id } = this.props;
    const currentCaseCondition = {
      case_id: channel_id,
      currentUserRole: this.props.userdata.user_type
    }
    const channelData = {
      channel_id: channel_id,
      type: "message"
    };
    this.props.setCurrentCase(currentCaseCondition);
    this.props.readMessages(channelData);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({ isLoading: true })
    if (nextProps.readedMessages !== this.props.readedMessages) {
      this.setState({
        readedMessages: nextProps.readedMessages,
        isLoading: false,
        readedMessagesWasSetted: true
      })
    }
    if (nextProps.currentCase !== this.props.currentCase) {
      this.setState({
        currentCase: nextProps.currentCase,
        isLoading: false,
        currentCaseWasSetted: true
      })
    }
  }

  callDownloadPDF = () => {
    this.child.downloadPdf();
  }

  render() {
    const {
      currentCase,
      readedMessages,
      userdata,
      isLoading,
      readedMessagesWasSetted,
      currentCaseWasSetted
    } = this.state;
    if (!isLoading && readedMessagesWasSetted && currentCaseWasSetted) {
      return (
        // <PDFViewer style={{ width: "100%", height: "100%" }}>
        <MyDocument
          userData={userdata}
          currentCase={currentCase}
          readedMessages={readedMessages}
          onRef={ref => (this.child = ref)}
        />
        // </PDFViewer>
      );
    } else {
      return (
        <div>
          <CircularProgress
            disableShrink
            style={{
              zIndex: 100,
              position: "relative",
              top: "-50vh",
              left: "50%"
            }}
          />
        </div>
      )
    }
  }
}

PDF.propTypes = {
  userdata: PropTypes.object.isRequired,
  readedMessages: PropTypes.array.isRequired,
  readMessages: PropTypes.func.isRequired,
  setCurrentCase: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  userdata: state.auth.userdata,
  currentCase: state.case.currentCase,
  readedMessages: state.chat.readMessages,
})

export default connect(
  mapStateToProps,
  { readMessages, setCurrentCase }
)(PDF)