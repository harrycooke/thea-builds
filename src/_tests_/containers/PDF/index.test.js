import React from 'react';
import ReactDOM from'react-dom';
import TestRenderer from 'react-test-renderer'; // ES6
import Provider from 'react-redux';
import sinon from 'sinon';

import {shallow, configure, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import {PDF} from '../../../containers/PDF/index';
import {callDownloadPDF} from '../../../containers/PDF/index';

describe('PDF', () => {
    beforeAll = () => {
        this.state = {
            channel_id: '',
            currentCase: '',
            readedMessages: [],
            isLoading: true,
            readedMessagesWasSetted: false,
            currentCaseWasSetted: false
        }
        this.callDownloadPDF = this.callDownloadPDF.bind(this)
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(PDF, div);
    });

    it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
        const renderer = TestRenderer.create(PDF);
        renderer.toJSON();
      });    

    describe('PDF', () => {
        it('should render correctly in "debug" mode', () => {
          const component = (PDF);
          expect(component).toMatchSnapshot();
        });
      });

    // it('calls funcName callDownloadPDF', () => {
    //     sinon.spy(PDF.callDownloadPDF);
    //     const wrapper = mount(<PDF />);
    //     expect(PDF.prototype.componentDidMount).to.have.property('callCount', 1);
    //   });
    
    //   it('allows us to set props', () => {
    //     const wrapper = mount(<PDF bar="baz" />);
    //     expect(wrapper.props().bar).to.equal('baz');
    //     wrapper.setProps({ bar: 'foo' });
    //     expect(wrapper.props().bar).to.equal('foo');
    //   });
    
    //   it('simulates click events', () => {
    //     const onButtonClick = sinon.spy();
    //     const wrapper = mount((
    //       <PDF onButtonClick={onButtonClick} />
    //     ));
    //     wrapper.find('button').simulate('click');
    //     expect(onButtonClick).to.have.property('callCount', 1);
    //   });
});

test('Fake test', ()=>{
    expect(true).toBeTruthy();          //here toBeTruthy is a predifined method whichif receives "true", considers to be expected.
});