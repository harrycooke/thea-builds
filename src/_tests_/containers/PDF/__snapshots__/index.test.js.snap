// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDF PDF should render correctly in "debug" mode 1`] = `undefined`;
